src/Lean/Meta/Tactic/AC/Main.lean

```diff
@@ -312,19 +312,30 @@ partial def rewriteUnnormalized (mvarId : MVarId) : M Unit :=
         assignExprMVar mvarId proof
       else throwError ""
     catch _ => throwError "cannot synthesize proof:\n{MessageData.ofGoal mvarId}"
-  | ProofStrategy.rfl tgt =>
-    trace[Meta.AC] "picking rfl strategy {MessageData.ofGoal mvarId}"
-    assignExprMVar mvarId (←mkAppM ``Eq.refl #[tgt])
+  | ProofStrategy.simp =>
+    trace[Meta.AC] "picking simp strategy {MessageData.ofGoal mvarId}"
+    let simpCtx ← Simp.Context.mkDefault
+    let newGoal ← simpTarget mvarId simpCtx
```

Comment:
Note that `simpTarget` returns `none` when the goal has been closed (a good thing). 
It confuses me from time to time. @Kha and I talked about adding a new type (with a `closed` constructor), but didn't find a good name.

Reply:
I am aware of that. It's just that the `simp` strategy is a finisher. So if it returns a new goal, `ac_refl` gives up and throws.

Reply:
done